apps/backend/charity/src/createCharity.ts

```diff
@@ -0,0 +1,100 @@
+import { Response } from "express";
+import { IDatabase } from "pg-promise";
+import { AuthenticatedRequest } from "../../lib/src/auth";
+
+interface Organization {
+  name: string;
+  logoUrl: string;
+  donated: number;
+  receiving: boolean;
+}
+
+export const createCharity = async (
+  req: AuthenticatedRequest,
+  res: Response,
+  db: IDatabase<object>,
+) => {
+  try {
+    const charity = req.body;
+    const { name, description, startDate, endDate, imageUrl, organizations } =
+      charity;
+
+    if (
+      !name ||
+      !description ||
+      !startDate ||
+      !endDate ||
+      !imageUrl ||
+      !organizations
+    ) {
+      console.error("missing parameter in request");
+      return res.status(400).json({ error: "missing parameter in request" });
+    }
+
+    for (const org of organizations) {
+      if (
+        !org.name ||
+        !org.logoUrl ||
+        org?.donated === null ||
+        org?.donated === undefined ||
+        org?.receiving === null ||
+        org?.receiving === undefined
+      ) {
+        console.error("missing parameter in request");
+        return res.status(400).json({ error: "missing parameter in request" });
+      }
+    }
+
+    const createdCharity = await db.oneOrNone(
+      `INSERT INTO charities (name, description, start_date, end_date, image_url) 
+      VALUES ($1, $2, $3, $4, $5) 
+      RETURNING charity_id, name, description, start_date, end_date, image_url`,
+      [name, description, startDate, endDate, imageUrl],
+    );
+
+    const queries = organizations.map((org: Organization) => {
+      return db.oneOrNone<Organization>(
+        `INSERT INTO organizations (name, logo_url, donated, receiving, charity_id)
+        VALUES ($1, $2, $3, $4, $5)
+        RETURNING name, logo_url, donated, receiving;`,
+        [
+          org.name,
+          org.logoUrl,
+          org.donated,
+          org.receiving,
+          createdCharity.charity_id,
+        ],
+      );
+    });
+
+    const results = await db.tx((t) => {
+      return t.batch(queries);
+    });
```

Comment:
This won't execute the queries on a single connection. Note that in [the documentation](https://vitaly-t.github.io/pg-promise/Database.html#task) the queries are done within the callback and not on the main database. So essentially move the map inside the callback and replace `db` with `t`.

Nit: from a db performance perspective I'd prefer `task` instead of `tx` since I don't think we need the safety guarantees of a transaction.